cmds/core/tail/tail.go

```diff
@@ -141,9 +144,11 @@ func readLastLinesBackwards(input readAtSeeker, writer io.Writer, numLines uint)
 	}
 	// find the start of the n-th to last line
 	data := lastNLines(readData, numLines)
-	// write the requested lines to the writer
-	if _, err = writer.Write(data); err != nil {
-		return err
+	// write the requested lines to the writer if any
+	if len(data) > 0 {
```

Comment:
This test will break Plan 9. On Plan 9, writing 0 length data is allowed, and is used, and will even result in a 0-length 9p TWRITE. That 0-length TWRITE results in a 0-length read(), and that in turn is why Plan 9 has never needed a shutdown system call. Sometimes, you WANT a 0-length write to go all the way through the system. The inventors of Unix learned this lesson, and applied it in Plan 9. Unix and its descendants never got the memo ;-)

From a philosophy standpoint, this kind of test probably should never be in a single program anyway, it can result in one program behaving differently from all others on different kernels. I've debugged that kind of thing and it is no fun at all ...

I think we should not do this. Thanks.

Reply:
Oh, that's nice to know :) I will remove that if.